fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexScrubberTest.java

```diff
@@ -313,8 +313,8 @@ void testScrubberInvalidIndexState() {
         openSimpleMetaData(hook2);
         buildIndexClean(nonValueIndex);
         try (OnlineIndexScrubber indexScrubber = newScrubberBuilder(nonValueIndex).build()) {
```

Comment:
I am not sure that I understand your meaning (is that a general purpose dry run & report operation?). Would it make sense to write a design document for this idea?

Reply:
Not dry-run. A test framework. So it might be something like:
```java
public abstract class BaseScrubbingTests {
    // some abstract methods to setup the index
    @Test
    void empty() {
        // don't save anything, just initialize the store
        // run scrubbing of various types
        // should report nothing, and should not error, other than potentially unsupported scrubbing type
    }

    @Test
    void missing() {
        // save some initial data
        // disable the index
        // save another record
        // force mark the index readable
        // scrub, should find a missing entry
    }
    
    @Test
    void dangling() {
        // save some initial data
        // disable the index
        // delete one of the records
        // force mark the index readable
        // scrub, should find a dangling entry
    }

    // ... more tests that can be agnostic to the actual index implementation
}
```

Then extend that for value indexes, lucene indexes, etc.